templates/strict_authoritative/bind/named.conf.local.j2

```diff
@@ -0,0 +1,119 @@
+//
+
+// Consider adding the 1918 zones here, if they are not used in your
+// organization
+//include "/etc/bind/zones.rfc1918";
+
+{% if bind9_statistics_enabled %}
+
+statistics-channels {
+  inet 127.0.0.1 port 8053 allow { 127.0.0.1; };
+};
+{% endif %}
+{% if bind9_masters|default() %}
+{%   for master in bind9_masters %}
+masters "{{ master.name }}" {
+{%     for addr in master.addresses %}
+    {{ addr }};
+{%     endfor %}
+};
+{%   endfor %}
+{% endif %}
+{% if bind9_masters_extra|default() %}
+{%   for master in bind9_masters_extra %}
+masters "{{ master.name }}" {
+{%     for addr in master.addresses %}
+    {{ addr }};
+{%     endfor %}
+};
+{%   endfor %}
+{% endif %}
+{% if bind9_acl is defined %}
+
+	// Custom acls
+{%   for acl_item in bind9_acl %}
+acl "{{ acl_item.name }}" {
+{%     for acl_address in acl_item.addresses %}
+	{{ acl_address }};
+{%     endfor %}
+};
+{%   endfor %}
+{% endif %}
+
+// The following zones are managed by this DNS Server //
+{% for zone in (bind9_zones_static + bind9_zones_dynamic)|sort(attribute='name') %}
+{% set zone_type = zone.type|default(bind9_zone_type|default('master')) %}
+zone "{{ zone.name }}" {
+    type {{ zone_type }};
+{%   if zone_type == 'master' %}
+    file "/etc/bind/zones/db.{{ zone.name }}";
+    allow-query {
+{%     if zone.allow_query is defined and zone.allow_query | length() > 0  %}
+{%       for allow_query_item in zone.allow_query | list %}
+        {{ allow_query_item }};
+{%       endfor %}
+    };
+{%     else %}
+        any;
+    };
+{%     endif %}
+{%     set zone_secondaries = zone.secondaries | default( bind9_secondaries | default( [] ) ) %}
+{%     set zone_allow_transfer = zone.allow_transfer | default( zone_secondaries + zone.also_allow_transfer | default( bind9_also_allow_transfer | default( [] ) ) ) %}
+{%     if zone_allow_transfer %}
+
+    // allow transfer from secundaries and extra also_allow_transfer hosts
+    allow-transfer {
+{%       for allow_transfer_item in zone_allow_transfer %}
+        {{ allow_transfer_item }};
+{%       endfor %}
+    };
+{%     endif %}
+{%     if bind9_notify_explicit %}
+    notify explicit;
+{%     else %}
+     notify {{ zone.notify | default(true) | ternary ('yes','no') }};
+{%     endif %}
+{%     set zone_also_notify = zone.also_notify | default( bind9_also_notify | default() ) %}
+{%     if zone_also_notify %}
+    also-notify {
+{%       for also_notify_item in zone_also_notify %}
+        {{ also_notify_item }};
+{%       endfor %}
+    };
+{%     endif %}
+{%     if (bind9_dnssec or zone.dnssec | default() ) and zone.dnssec | default( bind9_dnssec_zones_default_enabled ) %}
+    auto-dnssec maintain;
+    inline-signing yes;
+{%     endif %}
+{%     if zone.update_policy_grant | default() %}
+    update-policy {
+        grant {{ zone.name }}_ddns_update {{ zone.update_policy_grant }};
+    };
+{%     endif %}
+{%   elif zone_type == 'slave' %}
+    file "/var/lib/bind/db.{{ zone.name }}";
```

Comment:
A slave zone could also have `allow-query` and `allow-transfer` directives configured. We need it for our NS server **Massera**.